port_ocean/context/event.py

```diff
@@ -28,11 +44,39 @@ class EventType:
 class EventContext:
     event_type: str
     trigger_type: TriggerType = "machine"
-    resource_config: Optional["ResourceConfig"] = None
     attributes: dict[str, Any] = field(default_factory=dict)
+    _aborted: bool = False
     _port_app_config: Optional["PortAppConfig"] = None
     _parent_event: Optional["EventContext"] = None
     _event_id: str = field(default_factory=lambda: str(uuid4()))
+    _on_abort_callbacks: list[AbortCallbackFunction] = field(default_factory=list)
```

Comment:
This class starts to maintain multiple configurations and scenarios might be worth adding documentation to the class